github/repos_contents.go

```diff
@@ -119,6 +120,10 @@ func (s *RepositoriesService) DownloadContents(owner, repo, filepath string, opt
 	return nil, fmt.Errorf("No file named %s found in %s", filename, dir)
 }
 
+func escapeURLPath(path string) string {
+	return template.URLQueryEscaper(path)
```

Comment:
okay, seems to work well on a live URLs: 

 - https://api.github.com/repos/google-test/wjn1/contents/p%23%3F%25%2F%E4%B8%AD.go
 - https://api.github.com/repos/google-test/wjn1/contents/%24%26%2B%2C%3A%3B%3D%3F%40.txt

Several of the escapes are unnecessary, but none seem to cause any problems.